kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller_test.go

```diff
@@ -275,20 +275,223 @@ var _ = Describe("TaskRunToolCall Controller", func() {
+				},
+			})
+			defer teardown()
+
+			By("reconciling the trtc against an approval-rejecting HumanLayer client")
+
+			reconciler, _ := reconciler()
+
+			reconciler.MCPManager = &MockMCPManager{
+				NeedsApproval: true,
+			}
+
+			reconciler.HLClientFactory = &humanlayer.MockHumanLayerClientFactory{
+				ShouldFail:            false,
+				StatusCode:            200,
+				ReturnError:           nil,
+				ShouldReturnRejection: true,
+			}
+
+			result, err := reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      trtc.Name,
+					Namespace: trtc.Namespace,
+				},
+			})
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result.Requeue).To(BeFalse())
+
+			By("checking the taskruntoolcall status is set to ToolCallRejected")
+			updatedTRTC := &kubechainv1alpha1.TaskRunToolCall{}
+			err = k8sClient.Get(ctx, types.NamespacedName{
+				Name:      trtc.Name,
+				Namespace: trtc.Namespace,
+			}, updatedTRTC)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(updatedTRTC.Status.Phase).To(Equal(kubechainv1alpha1.TaskRunToolCallPhaseFailed))
+			Expect(updatedTRTC.Status.Status).To(Equal(kubechainv1alpha1.TaskRunToolCallStatusTypeToolCallRejected))
+			Expect(updatedTRTC.Status.StatusDetail).To(ContainSubstring("Tool execution rejected"))
+		})
+	})
+
+	Context("ReadyToExecuteApprovedTool -> Succeeded", func() {
+		It("transitions from ReadyToExecuteApprovedTool to Succeeded when a tool is executed", func() {
+			trtc, teardown := setupTestApprovalResources(ctx, &SetupTestApprovalConfig{
+				TaskRunToolCallStatus: &kubechainv1alpha1.TaskRunToolCallStatus{
+					ExternalCallID: "call-ready-to-execute-test",
+					Phase:          kubechainv1alpha1.TaskRunToolCallPhasePending,
+					Status:         kubechainv1alpha1.TaskRunToolCallStatusTypeReadyToExecuteApprovedTool,
+					StatusDetail:   "Ready to execute tool, with great vigor",
+					StartTime:      &metav1.Time{Time: time.Now().Add(-1 * time.Minute)},
+				},
+			})
+			defer teardown()
+
+			By("reconciling the trtc against an approval-granting HumanLayer client")
+
+			reconciler, _ := reconciler()
+
+			result, err := reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      trtc.Name,
+					Namespace: trtc.Namespace,
+				},
+			})
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result.Requeue).To(BeFalse())
+
+			By("checking the taskruntoolcall status is set to Ready:Succeeded")
+			updatedTRTC := &kubechainv1alpha1.TaskRunToolCall{}
+			err = k8sClient.Get(ctx, types.NamespacedName{
+				Name:      trtc.Name,
+				Namespace: trtc.Namespace,
+			}, updatedTRTC)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(updatedTRTC.Status.Phase).To(Equal(kubechainv1alpha1.TaskRunToolCallPhaseSucceeded))
+			Expect(updatedTRTC.Status.Status).To(Equal(kubechainv1alpha1.TaskRunToolCallStatusTypeReady))
+			Expect(updatedTRTC.Status.Result).To(Equal("5")) // From our mock implementation
```

Comment:
My #1 question reading this test...what are we testing? I don't see where we check "did the tool run"

I'm assuming its here, but where is this "5" set?  is it buried in the setup? If the CORE thing on this test is to make sure we ran the tool, I'd expect that setup to show up in this test body